vavr/src/test/java/org/jdbi/v3/vavr/TestVavrTupleMapperWithDB.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.vavr;
+
+import io.vavr.Tuple;
+import io.vavr.Tuple1;
+import io.vavr.Tuple2;
+import io.vavr.Tuple3;
+import io.vavr.collection.List;
+import org.jdbi.v3.core.generic.GenericType;
+import org.jdbi.v3.core.result.ResultSetException;
+import org.jdbi.v3.core.rule.H2DatabaseRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestVavrTupleMapperWithDB {
+
+    @Rule
+    public H2DatabaseRule dbRule = new H2DatabaseRule().withPlugins();
+
+    private List<Integer> expected = List.range(0, 9);
+
+    @Before
+    public void addData() {
+        dbRule.getSharedHandle().execute("create table tuples (" +
+                "t1 int, " +
+                "t2 varchar(10), " +
+                "t3 varchar(255), " +
+                ")");
+        for (Integer i : expected) {
+            dbRule.getSharedHandle().execute("insert into tuples values (?, ?, ?)",
+                    i, "t2" + i, "t3" + (i + 1));
+        }
+    }
+
+    @Test
+    public void testMapToTuple1_shouldSucceed() {
+        Tuple1<String> tupleProjection = dbRule.getSharedHandle()
+                .createQuery("select t2 from tuples order by t1 asc")
+                .mapTo(new GenericType<Tuple1<String>>() {
+                })
+                .findFirst().get();
```

Comment:
Maybe - did not occur to me to use `findOnly()` since i have a result set size of 10. i will look into it but probably assumed it would throw an exception.

Reply:
Ah, ok. If there are multiple results then `findFirst()` is what you want here